pipit/trace.py

```diff
@@ -190,6 +190,78 @@ def _match_events(self):
 
             self.events = self.events.astype({"_matching_event": "Int32"})
 
+    def _match_messages(self):
+        """
+        Matches corresponding MpiSend/MpiRecv and MpiIsend/MpiIrecv instant events
+        """
+        if "_matching_event" not in self.events.columns:
+            self.events["_matching_event"] = None
+
+        if "_matching_timestamp" not in self.events.columns:
+            self.events["_matching_timestamp"] = np.nan
+
+        matching_events = list(self.events["_matching_event"])
+        matching_times = list(self.events["_matching_timestamp"])
+
+        mpi_events = self.events[
+            self.events["Name"].isin(["MpiSend", "MpiRecv", "MpiIsend", "MpiIrecv"])
+        ]
+
+        queue = [[] for _ in range(len(self.events["Process"].unique()))]
```

Comment:
Update queue to be a dict per process, and index the dict per receiver process with sender processes. Also comments should be added to explain this.